gtsam/nonlinear/tests/testValues.cpp

```diff
@@ -589,6 +589,23 @@ TEST(Values, MatrixDynamicInsertFixedRead) {
   CHECK_EXCEPTION(values.at<Matrix23>(key1), exception);
 }
 
+TEST(Values, Demangle) {
+  Values values;
+  Matrix13 v; v << 5.0, 6.0, 7.0;
+  values.insert(key1, v);
+  string expected = "Values with 1 values:\nValue v1: (Eigen::Matrix<double, 1, 3, 1, 1, 3>) [\n	5, 6, 7\n]\n\n";
+
+  stringstream buffer;
+  streambuf * old = cout.rdbuf(buffer.rdbuf());
+
+  values.print();
+
+  string actual = buffer.str();
+  cout.rdbuf(old);
+
+  EXPECT(assert_equal(expected, actual));
```

Comment:
`EXPECT(assert_equal())`  sounds weird to me, but I'm not so familiar with this unit test suite. From GTest, I would expect something like EQUAL_STR(), but if you are sure that's the correct way, forget my comment

Reply:
I looked through other tests in GTSAM which do string comparisons and this seemed to be the universal way of doing it. But I know what you mean! Frank is very well aware of some of my frustrations with CppUnitLite.